### PR TITLE
doc: topology-over-raft: add transition_state to node state diagram

### PR DESCRIPTION
**Pull request by tchaikov**
in order to help the developers to understand the transitions of `node_state` and the `transition_state` on each of the `node_state`, in this change, the nested state machine diagram is added to the node state diagram.

please note, instead of trying to merge similar states like bootstrapping and replacing into a single state, we keep them as separate ones, and replicate the nested state machine diagram in them as well, to be more clear.

### CODE REVIEW

**Comment by tchaikov** [comment]
the rendered diagram can be found at https://github.com/scylladb/scylladb/blob/23370fa28cf5fcff73150e1ac2046181a3ddfa56/docs/dev/topology-over-raft.md

**Comment by tchaikov** [comment]
the rendered diagram can be found at https://github.com/scylladb/scylladb/blob/5570df65410292acc8d39f365aa3ef985ee394b6/docs/dev/topology-over-raft.md

File: docs/dev/topology-over-raft.md [hunk 1]
```diff
@@ -22,15 +22,81 @@ Nodes in state `left` may still appear as tablet replicas in host_id-based repli
...
+    none --> bootstrapping: join
+    none --> replacing: replace
+    state bootstrapping {
+        bs_join_group0 : join_group0
+        bs_left_token_ring : left_token_ring
+        bs_commit_cdc_generation : commit_cdc_generation
+        bs_write_both_read_old : write_both_read_old
+        bs_write_both_read_new : write_both_read_new
+        [*] --> bs_join_group0
+        bs_join_group0 --> bs_left_token_ring: rollback
```
Comment (gleb-cloudius) [inline docs/dev/topology-over-raft.md:34]: rollback can be entered from commit_cdc_generation and write_both_read_old as well.

File: docs/dev/topology-over-raft.md [hunk 2]
```diff
@@ -22,15 +22,81 @@ Nodes in state `left` may still appear as tablet replicas in host_id-based repli
...
+        bs_left_token_ring --> [*]
+    }
+    state replacing {
+        rp_join_group0 : join_group0
+        rp_left_token_ring : left_token_ring
+        rp_tablet_draining : tablet_draining 
+        rp_write_both_read_old : write_both_read_old
+        rp_write_both_read_new : write_both_read_new
+        [*] --> rp_join_group0
+        rp_join_group0 --> rp_left_token_ring: rollback
```
Comment (gleb-cloudius) [inline docs/dev/topology-over-raft.md:49]: Here as well (and same for all operations) rollback can be entered from write_both_read_old

**Comment by tchaikov** [comment]
v2:

- add the missing transitions triggered by rollback

**Comment by tchaikov** [comment]
before this change, https://github.com/scylladb/scylladb/blob/master/docs/dev/topology-over-raft.md

after this change, the rendered diagram can be found at https://github.com/scylladb/scylladb/blob/6d3e504525e330ac617e7ec49b9c4641a9a39579/docs/dev/topology-over-raft.md

File: docs/dev/topology-over-raft.md [hunk 3]
```diff
@@ -22,15 +22,86 @@ Nodes in state `left` may still appear as tablet replicas in host_id-based repli
...
+        rp_join_group0 --> rp_tablet_draining
+        rp_tablet_draining --> rp_write_both_read_old
```
Comment (gleb-cloudius) [inline docs/dev/topology-over-raft.md:53]: We can rollback from tablet_draining while replacing as well. Sorry missed it before.

Comment (tchaikov) [reply docs/dev/topology-over-raft.md:53]: thanks for pointing this out. added.

**Comment by tchaikov** [comment]
v3:

- add rollback path from `tablet_draining` in `replacing` state

**Comment by tchaikov** [comment]
before this change, https://github.com/scylladb/scylladb/blob/master/docs/dev/topology-over-raft.md

after this change, the rendered diagram can be found at https://github.com/scylladb/scylladb/blob/69b66223703eecf7580316294903bdec308fffcd/docs/dev/topology-over-raft.md

**Review by gleb-cloudius — APPROVED** [review]

**Review by kbr-scylla — APPROVED** [review]

**Pull request closed without merging**